proxy/api/src/http/session.rs

```diff
@@ -48,25 +48,29 @@ fn update_settings_filter(
 mod handler {
     use warp::{http::StatusCode, reply, Rejection, Reply};
 
-    use crate::{context, session};
+    use crate::{context, error, session};
 
     /// Clear the current [`session::Session`].
-    pub async fn delete(ctx: context::Context) -> Result<impl Reply, Rejection> {
+    pub async fn delete(ctx: context::Unsealed) -> Result<impl Reply, Rejection> {
         session::clear_current(&ctx.store)?;
 
         Ok(reply::with_status(reply(), StatusCode::NO_CONTENT))
     }
 
     /// Fetch the [`session::Session`].
     pub async fn get(ctx: context::Context) -> Result<impl Reply, Rejection> {
-        let sess = session::current(ctx.state.clone(), &ctx.store).await?;
-
-        Ok(reply::json(&sess))
+        match ctx {
```

Comment:
Why isn't this taking `Unsealed` and using the guard?

Reply:
Excellent catch! The code here is indeed a copy `with_context_unsealed`. I’ve pushed a fix